pomodoro/src/main/java/pomodoro/service/TeamService.java

```diff
@@ -30,24 +39,46 @@
         return teamsDto;
     }
 
+    @Transactional
+    @SuppressWarnings("unchecked")
+    public List<TeamDto> findUserTeams(Principal principal) {
+        Map<String, String> details = (Map<String, String>) ((OAuth2Authentication) principal).getUserAuthentication()
+                .getDetails();
+        User user = userRepository.findOne(details.get("email"));
+        List<TeamDto> teamsDto = new ArrayList<>();
+        for (Team team : user.getTeams()) {
+            TeamDto teamDto = new TeamDto();
+            teamDto.setTeamId(team.getTeamId());
+            teamDto.setName(team.getName());
+            teamsDto.add(teamDto);
+        }
+        return teamsDto;
+    }
+
     @Transactional(readOnly = true)
-    public TeamDto getById(Long teamId) {
+    public TeamDetailDto getById(Long teamId) {
         Team team = teamRepository.findOne(teamId);
-        TeamDto teamDto = new TeamDto();
-        teamDto.setTeamId(team.getTeamId());
-        teamDto.setName(team.getName());
-        return teamDto;
+        TeamDetailDto teamDetailDto = new TeamDetailDto(team);
+        return teamDetailDto;
     }
 
     @Transactional
-    public Team saveOrUpdate(TeamDto teamDto) {
-        Team team = teamRepository.findOne(teamDto.getTeamId());
+    @SuppressWarnings("unchecked")
+    public void saveOrUpdate(TeamDto teamDto, Principal principal) {
+        Team team = null;
+        if (teamDto.getTeamId() != null) {
+            team = teamRepository.findOne(teamDto.getTeamId());
+        }
         if (team == null) {
             team = new Team();
         }
-        team.setTeamId(teamDto.getTeamId());
         team.setName(teamDto.getName());
-        return teamRepository.save(team);
+        teamRepository.save(team);
+        Map<String, String> details = (Map<String, String>) ((OAuth2Authentication) principal).getUserAuthentication()
+                .getDetails();
+        User user = userRepository.findOne(details.get("email"));
+        user.getTeams().add(team);
+        userRepository.save(user);
```

Comment:
The relation between the team and a a user should be saved in a cascade, without the need to have to invoke a save method on both user or repository. It should either be `userRepository.save(user)`, and have the team saved in a cascade, or add a user to the team, and have the `teamRepository.save(team)`, also save the relation.